tanglescope/common/tangledb.cpp

```diff
@@ -20,4 +21,29 @@ std::optional<TangleDB::TXRecord> TangleDB::find(const std::string& hash) {
 void TangleDB::put(const TXRecord& tx) {
   std::unique_lock<std::shared_mutex> lock(mutex_);
   _txs[tx.hash] = tx;
-}
\ No newline at end of file
+}
+
+void TangleDB::removeAgedTxs(uint32_t ageInSeconds) {
+  auto now = std::chrono::system_clock::now();
+
+  std::unique_lock<std::shared_mutex> lock(mutex_);
+
+  std::vector<decltype(_txs)::key_type> vec;
+  for (auto&& it : _txs) {
+    if (std::chrono::duration_cast<std::chrono::seconds>(now -
+                                                         it.second.timestamp)
+            .count() > ageInSeconds) {
+      vec.emplace_back(it.first);
```

Comment:
can't you just remove in-place?

Reply:
that will invalidate the map it

Reply:
http://en.cppreference.com/w/cpp/algorithm/remove